blog/content/post/2024-08-19-secureboot-freebsd.md

```diff
@@ -0,0 +1,397 @@
+---
+title: Secure Booting FreeBSD with Dasharo firmware
+abstract: 'Abstract first sentence.
+          Abstract second sentence.
+          Abstract third sentence.'
+cover: /covers/image-file.png
+author: filip.lewinski
+layout: post
+published: true
+date: 2024-10-14
+archives: "2024"
+
+tags:
+  - bsd
+  - dasharo
+  - edk2
+  - secure-boot
+  - protectli
+  - uefi
+categories:
+  - Firmware
+  - IoT
+  - Miscellaneous
+  - OS Dev
+  - App Dev
+  - Security
+  - Manufacturing
+
+---
+
+## Introduction
+
+## UEFI Secure Boot
+
+UEFI Secure Boot is a security feature designed to ensure that only trusted,
+signed software can execute during the boot process, protecting against malware
+and unauthorized code execution. It verifies the signatures of bootloaders and
+kernel binaries against trusted certificates embedded in the firmware.
+
+### Linux
+
+Most distributions took the approach of using a small, signed bootloader called
+`shim`, which is pre-signed by Microsoft. Shim loads the bootloader, which
+is verified against a certificate embedded in shim, and the bootloader then
+loads the Linux kernel, also ensuring it is signed and trusted. This way, they
+can sign their kernel themselves, instead of having to ask Microsoft to sign
+each kernel update.
+
+Please see [this blogpost](https://mjg59.dreamwidth.org/20303.html) for more
+information on this specific subject.
+
+### UEFI Secure Boot status on FreeBSD - brief summary
+
+In short, the [FreeBSD wiki](https://wiki.freebsd.org/SecureBoot) proposes two
+stages of Secure Boot implementation.
+
+First would be to enable booting the system with Microsoft-signed shim and a
+FreeBSD-signed EFI loader, while also offering users to generate their own
+keys and certificates.
+
+The second stage would involve having all of the drivers and kernel modules to
+be signed and authorized, resulting in a locked-down system akin to Fedora's
+implementation.
+
+As of the time of writing this post, the first stage is mostly complete. It is
+possible to create and sign a complete EFI executable containing the bootloader
+and the FreeBSD kernel, which can be booted using Dasharo firmware Secure Boot.
```

Comment:
UEFI Secure Boot?

Reply:
[Corrected](https://github.com/3mdeb/news-and-ideas/pull/603/commits/469cc84541acf2428e46af01de19cf63b5bb0466#diff-1e142bc7519e97361fc0d3867abcb7fa3d257a4b5ccb79795e5e00acebf7d22fR68)